ghostery/ghostery-extension
# Renew onboarding

Fixes https://github.com/ghostery/journal/issues/116

### The feature highlights

* Added content script to the main bundle, so the popup is triggered if conditions are met
* Added background module, which initializes local storage data if conditions are met
* Added `renew` component in onboarding UI module
* Main onboarding screen updated with `renew` flag, which changes copy and `abort` action (without confirm dialog)

### Testing

1. Turn on `debug` mode in `manifest.json`, by adding `"debug": true`
2. Ensure the staging AB testing server returns `true` for `the "terms"` test for your browser
3. Install the extension and make a normal setup
4. Open DevTools on the background process (or any extension page), and run:
    `chrome.storage.local.set({ enable_human_web: false })`
5. Restart the browser (the background process must re-run)
6. Open any web page
7. You should see (only once per hour) a popup with a message about renewing the setup
